src/engraving/tests/exchangevoices_data/undoChangeVoice02-ref.mscx

```diff
@@ -209,6 +209,9 @@
               <tpc>15</tpc>
               </Note>
             </Chord>
+          <location>
+            <fractions>1/4</fractions>
+            </location>
```

Comment:
Could you explain why these changes occur?

Reply:
Those changes occur because of the original commit from the comminity PR. I'll take this as an example. The second voice has 1/4 note and 3/4 of gap rest. 
![image](https://github.com/user-attachments/assets/010c5721-2952-4aa8-b36f-23228277648e)
The gap rest isn't written in our files as an actual rest, but as a location move (which I don't like, but that's beside the point). In other words, we represent a 1/4 gap rest as a "move ahead by 1/4" instruction. _Except_, if the destination of the location move coincides with the end of the measure. In that case the location move isn't written because the start of the next measure resets the current tick anyway. 
Now, the difference occurs because previously the gap rest was forced to always be a single rest, which means the underlying gap looks like this 
![image](https://github.com/user-attachments/assets/3f5d2027-1f3c-4efd-a75f-813da203dfdb)
but that gap of 3/4 doesn't get written in the file because the destination coincides with the measure end.
Now, however, we break gaps rests just like normal rests, which means the gap rests now look like this. 
![image](https://github.com/user-attachments/assets/ff152727-bc6a-4b67-87a1-4681e626d86b)
Now, the 1/4 rest gets written (that's what creates this diffs in the file), and the 2/4 will be skipped because it's end of the measure.

Reply:
> _Except_, if the destination of the location move coincides with the end of the measure.

Ah, that explains it! 
Of course, one could have an opinion about how much sense it makes that this causes the quarter rest to be "written" and the half rest not, but I think it doesn't really matter, since the whole concept of gaps and location tags probably won't exist for long anymore anyway.